src/gui/tracks/TrackContentWidget.cpp

```diff
@@ -82,16 +95,27 @@ TrackContentWidget::TrackContentWidget( TrackView * parent ) :
 
 
 
-
-
-
 void TrackContentWidget::updateBackground()
 {
+	// use snapSize to determine number of lines to draw
+	float snapSize = getGUI()->songEditor()->m_editor->getSnapSize();
+
 	const TrackContainerView * tcv = m_trackView->trackContainerView();
 
 	// Assume even-pixels-per-bar. Makes sense, should be like this anyways
 	int ppb = static_cast<int>( tcv->pixelsPerBar() );
 
+	// Coarse grid appears every bar (less frequently if quantization > 1 bar)
+	// Fine grid handles smaller grids
+	float coarseGridResolution = (snapSize >= 1) ? snapSize : 1;
+	float fineGridResolution = snapSize;
+	// Decrease fine grid resolution if it results in less than
+	// 4 pixels between each line
+	while ((ppb * fineGridResolution) < MIN_PIXELS_BETWEEN_LINES)
```

Comment:
This can be computed directly instead of converging in a while loop.

Reply:
Just curious - is this for the sake of performance, or readability? I came up with something that doesn't use loops (and is probably faster) but might be less readable.

It looks like this, for reference:
```
float fineGridResolution = snapSize;

// Increase fine grid resolution (size between lines) if it results in less than	
// 4 pixels between each line
float pixelsBetweenLines = ppb * snapSize;
if (pixelsBetweenLines < MIN_PIXELS_BETWEEN_LINES) {
	// scaleFactor should be a power of 2
	int scaleFactor = 1 << static_cast<int>( std::ceil( std::log2( MIN_PIXELS_BETWEEN_LINES/pixelsBetweenLines ) ) );
	fineGridResolution *= scaleFactor;
}
```

Reply:
For me it's a bit of both, performance and readability. If the reference code would be improved by adding a short comment I would prefer it.